client/src/screens/groups/GroupDetails.js

```diff
@@ -325,36 +138,72 @@ class GroupDetails extends Component {
 
     // somehow we got here with no group
     if (!group) {
-      return <Text>Error displaying group</Text>;
+      return (
+        <Container>
+          <Center>
+            <Text>Nothing here. thats weird.</Text>
+          </Center>
+        </Container>
+      );
     }
 
-    const rows = [
-      ['info', group],
-    ].concat(
-      _.map(['Users'], r => ['section', r]),
-      _.map(group.users, r => ['user', r]),
-      _.map(['Events'], r => ['section', r]),
-      _.map(group.events, r => ['event', r]),
-      _.map(['Schedules'], r => ['section', r]),
-      _.map(group.schedules, r => ['schedule', r]),
-    );
-    // render group stuff
-
     const memberAlready = _.some(group.users, g => g.id === me.id);
+    const tags = group.tags.map(elem => `#${elem.name}`).join(', ');
 
     return (
       <Container>
-        <Button
-          title={memberAlready ? 'Leave Group' : 'Join Group'}
-          onPress={memberAlready ? this.leaveGroup : this.joinGroup}
+        <View style={styles.topContainer}>
+          <View style={styles.iconLeftHolder}>
+            <Icon style={styles.icon} name={group.icon} size={30} />
+          </View>
+          <View style={styles.textContainer}>
+            <Text style={styles.titleTextBold} numberOfLines={1}>
+              {group.name}
+            </Text>
+            <Text style={styles.subtitleText} numberOfLines={1}>
+              {tags !== '' ? tags : 'No Tags'}
+            </Text>
+          </View>
+          <TouchableOpacity
+            style={styles.iconRightHolder}
+            onPress={memberAlready ? this.leaveGroup : this.joinGroup}
+          >
+            <Icon style={styles.icon} name={memberAlready ? 'sign-out' : 'sign-in'} size={25} />
+            <Text style={styles.leaveGroupText}>{memberAlready ? 'Leave Group' : 'Join Group'}</Text>
+          </TouchableOpacity>
+        </View>
+        <Holder margin transparent>
+          <Segment
+            values={['Users', 'Schedules', 'Events']}
+            handleIndexChange={this.handleIndexChange}
+            selectedIndex={this.state.selectedIndex}
+            refetch={this.onRefresh}
+          />
+        </Holder>
+        {this.state.selectedIndex === 0 && (
+        <GroupDetailsList
```

Comment:
how about we make this component take the entire `group` object and then pick users, schedules or events based upon the `render` parameter (although I recommend `mode` for reasons specified above)? This way you only need to define GroupDetailsList once. I don't think it matters that you pass down props you're not going to use for each particular render - the cost of passing some references is minimal.
```
<GroupDetailsList
  group={group}
  mode={['users','schedules','events'][this.state.selectedIndex]}
  networkStatus={networkStatus}
  refetch={this.onRefresh}
/>
```

Reply:
worked around this in a smarter way